nmostesting/suites/IS1101Test.py

```diff
@@ -13,14 +13,336 @@
+            _, response = TestHelper.do_request(
+                "GET", self.compat_url + "outputs/" + output_id + "/edid/"
+            )
+            if response.status_code != 204:
+                return test.UNCLEAR("Status code should be 204.")
+        return test.PASS()
+
+    # RECEIVERS TESTS
+    def test_04_01(self, test):
+        """
+        Verify that the device supports the concept of IS-11 Receiver.
+        """
+        _, response = TestHelper.do_request("GET", self.compat_url + "receivers/")
+
+        if response.status_code != 200:
+            return test.FAIL("The request has not succeeded.", response.json())
+        self.receivers = response.json()
+        if len(self.receivers) == 0:
+            return test.UNCLEAR("No IS_11 receivers.")
+        return test.PASS()
+
+    def test_04_01_01(self, test):
+        """
+        Verify that IS-11 Receivers exist on the Node API as Receivers.
+        """
+        if len(self.receivers) == 0:
+            return test.UNCLEAR("No IS_11 receivers.")
+        for receiver_id in self.receivers:
+            _, response = TestHelper.do_request(
+                "GET", self.node_url + "receivers/" + receiver_id
+            )
+            if response.status_code != 200:
+                return test.FAIL("The request has not succeeded.", response.json())
+            if response.json()["id"] != receiver_id[:-1]:
+                return test.UNCLEAR(
+                    "The IS-11 Receiver doesn't exist on the Node API as receiver."
+                )
+        return test.PASS()
+
+    def test_04_02(self, test):
+        """
+        Verify receivers (generic with/without outputs)
+        """
+        _, response = TestHelper.do_request("GET", self.compat_url + "receivers/")
+        if response.status_code != 200:
+            return test.FAIL("The request has not succeeded.", response.json())
+        self.receivers = response.json()
+        if len(self.receivers) == 0:
+            return test.UNCLEAR("No IS_11 receivers.")
+        return test.PASS()
+
+    def test_04_02_01(self, test):
+        """
+        Verify that the status is "unknown" or "non_compliant_stream"
+        as per our pre-conditions of not being master_enabled.
+        """
+        if len(self.receivers) == 0:
+            return test.UNCLEAR("No IS_11 receivers.")
+        for receiver_id in self.receivers:
+            _, response = TestHelper.do_request(
+                "GET", self.compat_url + "receivers/" + receiver_id + "status/"
+            )
+            if response.status_code != 200:
+                return test.FAIL("The request has not succeeded.", response.json())
+            if response.json()["state"] not in ["unknown", "non compliant stream"]:
+                return test.FAIL("The state is not unknown or non compliant stream.")
+        return test.PASS()
+
+    def test_04_02_02(self, test):
+        """
+        Verify that the Receiver supports Receiver Capabilities.
```

Comment:
@alabou Which part of IS-11 suggests/recommends/requires that Receivers support Receiver Capabilities?

If it is optional, then correct test result if not present should be `OPTIONAL` (Not Implemented). If it is recommended, should be `WARNING` (Warning). If it is required, should be `FAIL` (Fail).

If it isn't mentioned or is currently only optional in the IS-11 spec, should it be made recommended or required?

Reply:
The  test plan says "Verify that the Receiver supports Receiver Capabilities. The IS-11 specification indicates SHOULD use BCP-004-01 to express Receiver Capabilities. If a Receiver does not support BCP-004-01 we simply skip it **with a warning**."

Reply:
> The IS-11 specification indicates SHOULD use BCP-004-01 to express Receiver Capabilities.

In which section of IS-11 does it say that?

Reply:
https://specs.amwa.tv/is-11/branches/v1.0-dev/docs/Interoperability.html
Receivers, managed with IS-11 and associated with Outputs which support EDID, SHOULD use BCP-005-01 to express Receiver Capabilities that came from the EDID(s) and BCP-004-01 to express their own Receiver Capabilities.

Reply:
I saw that but that seems to only cover Receivers that are associated with Outputs that support EDID.

Reply:
I agree that the sentence is not quite clear ... it may be my brain that associate "associated with Outputs which support EDID, SHOULD use BCP-005-01 to express Receiver Capabilities that came from the EDID(s)" as one thing then "and BCP-004-01 to express their own Receiver Capabilities." for "Receivers managed with IS-11" as the general clause ...

IS-11 is based on Receiver Capabilities ... so it would be strange to least not have a SHOULD for BCP-004-01 ...

Reply:
Makes sense, and I agree. We can keep that meaning but maybe split into two sentences? In this test case then it should be `WARNING` not `UNCLEAR` if not present.

Reply:
Open issue to track this future change
https://github.com/AMWA-TV/nmos-testing/issues/792